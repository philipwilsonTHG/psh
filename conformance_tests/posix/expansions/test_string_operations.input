# Test string operations in parameter expansion

# Setup test variable
text="hello_world_test"
prefix="hello"
suffix="test"

# Test prefix removal
echo "Remove prefix 'hello': ${text#hello}"
echo "Remove prefix 'hello_': ${text#hello_}"

# Test suffix removal  
echo "Remove suffix 'test': ${text%test}"
echo "Remove suffix '_test': ${text%_test}"

# Test longest prefix removal
long_text="dir/subdir/file.txt.bak"
echo "Remove shortest path: ${long_text#*/}"
echo "Remove longest path: ${long_text##*/}"

# Test longest suffix removal
echo "Remove shortest extension: ${long_text%.txt.bak}"
echo "Remove longest extension: ${long_text%%.*}"

# Test with patterns
files="file1.txt file2.log file3.txt"
for file in $files; do
    echo "Base name of $file: ${file%.txt}"
done