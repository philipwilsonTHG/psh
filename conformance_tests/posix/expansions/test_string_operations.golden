Remove prefix 'hello': _world_test
Remove prefix 'hello_': world_test
Remove suffix 'test': hello_world_
Remove suffix '_test': hello_world
Remove shortest path: subdir/file.txt.bak
Remove longest path: file.txt.bak
Remove shortest extension: dir/subdir/file
Remove longest extension: dir/subdir/file
Base name of files: files
